### PR TITLE
Update Error description when plugin not installed

### LINKED ISSUE #5896: Confusing error message for `get_widget` when plugin name doesn't exist
## Problem
The `get_widget` function raises several different errors
https://github.com/napari/napari/blob/902f5197a9e6f4752c0c673a0f4e06b01b407081/napari/plugins/_plugin_manager.py#L583

But there is no specific error if the plugin *name*  doesn't exist.
It is raise in the same case than when the widget `_dock_widget` is `null`.

https://github.com/napari/napari/blob/902f5197a9e6f4752c0c673a0f4e06b01b407081/napari/plugins/_plugin_manager.py#L637

As a result, I find the error message confusing:
https://github.com/napari/napari/blob/902f5197a9e6f4752c0c673a0f4e06b01b407081/napari/plugins/_plugin_manager.py#L619

## Step to reproduce
```python
import napari
viewer = napari.Viewer()
viewer.window.add_plugin_dock_widget(
    plugin_name="napari_bug", widget_name="Bug Report"
)
```

```python
KeyError: "Plugin 'napari_bug' does not provide any dock widgets"
```

## 📚 Documentation

I suppose there is two solutions:
- Create a specific error message when `plg_wdgs` is empty because there no
    -  dock_widget in the plugin
    - plugin
- Adapt the error message to suggest both cases

### PR DESCRIPTION
**Pull request by ClementCaporal**
# Fixes/Closes

Closes #5896

# Description
This pull request clearifies the error message raised when the user attempts to add a dock widget to the viewer when the plugin is not installed.

# References

As described in https://github.com/napari/napari/issues/5896 with @kevinyamauchi.

## Type of change

- [X] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/plugins/_plugin_manager.py [hunk 1]
```diff
@@ -606,7 +606,7 @@ def get_widget(
         Raises
         ------
         KeyError
-            If plugin `plugin_name` does not provide any widgets
+            If plugin `plugin_name` is not installed or does not provide any widgets
```
Comment (kevinyamauchi) [inline napari/plugins/_plugin_manager.py]: Minor nitpick: might be nice to have a period at the end.

```suggestion
            If plugin `plugin_name` is not installed or does not provide any widgets.
```

**Review by kevinyamauchi — APPROVED** [review]
Thank you @ClementCaporal for the contribution! I think this looks good. I left a super minor nitpick below.

**Review by psobolewskiPhD — APPROVED** [review]
Thanks! I think this looks good. In the long run, we'll want to split this to account for each case separately but it's tricky now with npe/npe2 coexistance.
Yet another motivator to use the shim and drop npe? cc: @DragaDoncila

**Comment by kevinyamauchi** [comment]
> I think this looks good. In the long run, we'll want to split this to account for each case separately but it's tricky now with npe/npe2 coexistance.

I agree that we want to be able to differentiate the two cases in the future. However, it's definitely an improvement to be tell the user there are two possible reasons for the error. Since there are two approvals and it's sat for a couple of days, I will merge now!

Congrats, @ClementCaporal on your first code contribution to napari :)

**Pull request merged**